inception-recommendation-api/src/main/java/de/tudarmstadt/ukp/inception/recommendation/api/evaluation/EvaluationResult.java

```diff
@@ -17,28 +17,226 @@
+            aAnnotatedPairs.forEach(this::incConfusionMatrix);
+            numOfLabels = countLabels();
+        }
+    }
+    
+    private int countLabels()
+    {
+        if (ignoreLabels != null) {
+            return Math.toIntExact(labels.stream().filter(l -> !ignoreLabels.contains(l)).count());
+        }
+        return labels.size();
+    }
+
+    public int getNumOfLabels()
+    {
+        return numOfLabels;
+    }
+
+    public EvaluationResult(List<String> aIgnoreLabels, Stream<AnnotatedTokenPair> aAnnotatedPairs,
+            int aTrainSetSize, int aTestSetSize)
+    {
+        this(aIgnoreLabels, aAnnotatedPairs);
+        trainingSetSize = aTrainSetSize;
+        testSetSize = aTestSetSize;
+    }
+
+    private void incConfusionMatrix(AnnotatedTokenPair aPair)
+    {
+        String goldLabel = aPair.getGoldLabel();
+        String predictedLabel = aPair.getPredictedLabel();
+
+        // annotated pair is true positive
+        if (goldLabel.equals(predictedLabel)) {
+            incCounter(goldLabel, goldLabel);
+        }
+        else {
+            // annotated pair is false negative for gold class = annotated pair is false
+            // positive for predicted class
+            incCounter(goldLabel, predictedLabel);
+        }
+
+        if (ignoreLabels == null || !ignoreLabels.contains(goldLabel)) {
+            total += 1;
+        }
+    }
+
+    private void incCounter(String aGoldLabel, String aPredictedLabel)
+    {
+        initConfEntries(aGoldLabel, aPredictedLabel);
+        
+        int count = confusionMatrix.get(aGoldLabel).get(aPredictedLabel) + 1;
+        confusionMatrix.get(aGoldLabel).put(aPredictedLabel, count);
+    }
+
+    private void initConfEntries(String aGoldLabel, String aPredictedLabel)
+    {
+        Map<String, Integer> initMap;
+        if (!confusionMatrix.containsKey(aGoldLabel)) {
+            initMap = new HashMap<>();
+            initMap.put(aGoldLabel, 0);
+            confusionMatrix.put(aGoldLabel, initMap);
+
+            
+            labels.add(aGoldLabel);
+            
+        }
+        if (!confusionMatrix.get(aGoldLabel).containsKey(aPredictedLabel)) {
+            confusionMatrix.get(aGoldLabel).put(aPredictedLabel, 0);
+            
+            if (!labels.contains(aPredictedLabel)) {
+                labels.add(aPredictedLabel);
+                initMap = new HashMap<>();
+                initMap.put(aPredictedLabel, 0);
+                confusionMatrix.put(aPredictedLabel, initMap);
+            }
+        }
     }
 
     /**
-     * Set the specific score which the recommender evaluation uses e.g. accuracy or f-score.
+     * Calculate accuracy, ignoring the ignoreLabel class as a gold label.
+     * 
+     * @return accuracy score
+     */
+    public double computeAccuracyScore()
+    {
+        double tp = 0.0;
+        for (String label : labels) {
+            if (ignoreLabels == null || !ignoreLabels.contains(label)) {
```

Comment:
I would forbid `ignoreLabels ` being null, just use an empty set when constructing an instance of this